src/index.d.ts

```diff
@@ -0,0 +1,3 @@
+interface Window {
```

Comment:
Is there no way we can get this picked up from the cockpit repo?

Reply:
```
# No files matching the pattern "src/index.d.ts" were found.
# Please check for typing mistakes in the pattern.
```
So that is the real issue :)

Reply:
This is a bit of an annoying situation:

```
not ok 1 /static-code/test-eslint
#
# Oops! Something went wrong! :(
#
# ESLint: 8.57.0
#
# No files matching the pattern "src/index.d.ts" were found.
# Please check for typing mistakes in the pattern.
```

So this is an error from eslint, which I can't reproduce ofcourse with `npm run eslint` so the conclusion is that our cooked up `test/static-code` does the wrong thing.

Second issue is that TS fails:

```
not ok 10 /static-code/test-typescript
# pkg/lib/cockpit-upload-helper.ts(27,16): error TS2339: Property 'debugging' does not exist on type 'Window & typeof globalThis'.
# pkg/lib/cockpit-upload-helper.ts(27,45): error TS2339: Property 'debugging' does not exist on type 'Window & typeof globalThis'.
# src/upload-button.tsx(78,11): error TS2322: Type 'ReactElement<{}, string | JSXElementConstructor<any>>' is not assignable to type 'string'.
# src/upload-button.tsx(297,30): error TS2353: Object literal may only specify known properties, and '"--progress"' does not exist in type 'Properties<string | number, string & {}>'.
```

Reply:
I tried creating `pkg/lib/window.d.ts` or even `pkg/lib/global.d.ts` but neither get picked up when running:

```
tsc --checkJs false   --skipLibCheck 
```

So I don't see what else we can do here. It doesn't have anything to do with `--skipLibCheck` as leaving that out also fails.

Reply:
Spending an half hour on investigating this I will just leave index.d.ts and let someone else figure this mess out. Likely we should just move the typing into cockpit.d.ts as well it is more or less a Cockpit feature.